skrub/_agg_joiner.py

```diff
@@ -6,49 +6,140 @@
+    """Aggregate `table` on `key` and compute statistics on `cols_to_agg`.
 
-def split_num_categ_operations(operations):
-    """Separate aggregator operators input by their type.
+    Operations ["sum", "median", "mean", "std"] are only supported for numeric columns.
+
+    Add a suffix to columns not in `key` and sort columns by name.
 
     Parameters
     ----------
-    operations : list of str
-        The input operators names.
+    table : DataFrame
+        The input dataframe to aggregate.
+
+    key : str or iterable of str
+        The columns used as keys to aggregate on.
+
+    cols_to_agg : str or iterable of str
+        The columns to aggregate.
+
+    operations : str or iterable of str
+        The reduction functions to apply on columns
+        in ``cols_to_agg`` during the aggregation.
+
+    suffix : str
+        The suffix appended to output columns. Will only be applied
+        to columns created by the aggregations.
 
     Returns
     -------
-    num_operations, categ_operations : Tuple of list of str
-        List of operator names.
+    DataFrame
+        The aggregated output.
     """
-    num_operations, categ_operations = [], []
-    for operation in operations:
-        # hist(5) -> hist
-        op_root, _ = _parse_argument(operation)
-        if op_root in NUM_OPERATIONS:
-            num_operations.append(operation)
-        if op_root in CATEG_OPERATIONS:
-            categ_operations.append(operation)
-        if op_root not in ALL_OPS:
-            raise ValueError(f"operations options are {ALL_OPS}, got: {operation=!r}.")
 
-    return num_operations, categ_operations
+    # summing strings works in pandas, not in polars
+    num_only_operations = ["sum", "median", "mean", "std"]
+
+    key = atleast_1d_or_none(key)
+    cols_to_agg = atleast_1d_or_none(cols_to_agg)
+    operations = atleast_1d_or_none(operations)
+
+    table_to_agg = s.select(table, s.cols(*key) | s.cols(*cols_to_agg))
+
+    # Don't check the ID column, as it's not the one we aggregate on
+    table_to_check = s.select(table_to_agg, ~s.cols(*key))
+    not_numeric_cols = (~s.numeric()).expand(table_to_check)
+
+    num_only_op = list(set(operations).intersection(set(num_only_operations)))
+
+    if (len(not_numeric_cols) > 0) & (len(num_only_op) > 0):
+        raise AttributeError(
+            f"The operations {num_only_operations} are restricted to numeric columns."
+            f" \nConsider removing the following columns: {not_numeric_cols} or the"
+            f" following operations: {num_only_op}."
+        )
+
+    aggregated = perform_groupby(table_to_agg, key, cols_to_agg, operations)
+
+    new_col_names = [
+        f"{col}{suffix}" if col not in key else col
+        for col in sbd.column_names(aggregated)
+    ]
+    new_col_names = _join_utils.pick_column_names(new_col_names)
+    aggregated = sbd.set_column_names(aggregated, new_col_names)
+    aggregated = s.select(aggregated, sorted(sbd.column_names(aggregated)))
```

Comment:
I kept it because it was in the behavior of `aggregate`. Since operations were applied on columns in 2 steps (for value_counts / hist, and for all others), I assume this was done to put all the aggregation results on a column together.

In the new case I agree it could be unexpected, let's remove it